mantidproject/mantid
# Create ProfileChiSquared algorithm and clean up CalculateChiSquared

***Description of work***

This PR creates a profile Chi squared algorithm, which can be used to obtain better parameter error estimates from a fit. This is particularly useful if the function is non-linear, meaning the estimate from the covariance matrix may not be very accurate. It starts from an implementation within the CalculateChiSquared file, but adjusts for a few approximations. The algorithm has been compared to LMFIT and analytical calculations for the linear case - showing good agreement in both. 



**To test:**
In general, the algorithm should be run after a least-squares fit has been performed (and we are at a minimum).
- Load some data, create a fit function and Fit to the data/some of the data
- Run the algorithm
- Look at the errors and pdf table and check the numbers seem valid

I've been testing with the following scripts (which require LMFIT):
- Linear fit (where we can compare against an analytical result):
https://gist.github.com/StephenSmith25/e7f1bca3b730c571d9feea6ee1a2d721
- Nonlinear fit:
https://gist.github.com/StephenSmith25/736009368e6c7523f3d6699e9a0837a3


Fixes #30568 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
